### DISCUSSION TITLE
hamaの進捗状況

### DISCUSSION THREAD

**Discussion by hamanyann** [lgtm-factory/lgtm-factory#34] [Scrap]
全然理解が追い付かなくてissueに着手できていないのでこまめに進捗報告します。
noteにまとめてましたが、一日の最後にまとめて報告するよりDiscussionsでカジュアルに記録していくとのことだったのでここで書きます。
不要でしたら閉じます。

**Comment by hamanyann** [comment]
https://note.com/hamakigyou/n/n00431ff5be4a
8/7(水)　GitHubについて調べました。

**Comment by hamanyann** [comment]
今日もnoteにまとめていましたが、すでに解決したところは省略します。

**.md**

今回の自分のissueはCONTRIBUTING.mdに、issue ベースの開発の流れを記載すること。
mdはMarkdownファイル。
がっつり書くのは初めてなので調べながらやっていきます。

**その前にやらなくちゃいけないこと**

自分のissueを進める前にDiscussionをもう一度見直していきます。
日本語の理解力がないので、気合い入れて読んでいかないと理解出来ない。。

**isuueについて**

・バグの報告や新規機能追加の要求などに関する議論を行う
・完了していないタスクが掲載されるToDoリスト

ラベルを付けることで内容が分かりやすくなる。

デフォルトの機能かと思ったらこの緊急マークはオリジナルのラベルだったんですね。
今のところラベルはこの「優先ラベル」のみ。

Assignessが設定されてないissueに自分を割り当ててから作業を行う。
一つのissueは複数人で作業しない。

issueに着手している最中はDiscussionsを使って作業をチームメンバーに共有する。
終わったissueは閉じる。

**Comment by hamanyann** [comment]
**Discussionsについて**

議論、質問、作業メモ、方針や知見を共有する場。issueに着手する際、discussionに調べたこと、メモ、作業記録を残す。

メリット・タスクの進捗をチームに知らせる・チーム全員に知見を共有できる・作業の途中でも訂正・アドバイスが出来る・事前に作業内容がわかるとレビューする人の負担が減る

レビューする際に作業経過の共有があるとレビューが楽になる。簡単なタスクや細かい修正は共有不要。

**ブランチ戦略とは**

ブランチを作る際のルールを決めることで、無法地帯にならずに計画的に運用することが出来る。

GitHub Flowが定番であり、今回もGitHub Flowで行う。

**Comment by hamanyann** [comment]
> なぜ我々のチームがGitHub Flowを採用したかというと、Insight Edgeでは小回りの効く比較的小規模なチームで素早く開発をすることが多かったから、が理由になります。Git Flowのような厳密な管理をするブランチ戦略は、先の説明の通りリリースまでの工程や確認が多くなります。そのため、こまめに機能をリリースしながらシステムをアップデートしていくというアジャイル的な我々の開発スタイルにミートしませんでした。GitHub Flowのようにシンプルでこまめにリリースを重ねていける開発スタイルが小規模チームは適していると考えました。

何言ってるかわからな過ぎて心折れそうだけど読み解いていく。

**Reply by kazzyfrog** [reply]
そのコメントアウトで引用した記事の文章自体が、自分から見ても、少しわかりづらい？かもと思ったので、
手前味噌ですが、自分が以前書いた記事を、参考までに貼っておきます！
[参考: 2.GitHub Flow で手順書を作成する](https://zenn.dev/kazzyfrog/articles/github-guide#2.github-flow-で手順書を作成する)

そのコメントアウトで引用した記事の文章は、
要は、こんな感じの内容です！

- 最も有名な、ブランチ戦略が、「Git Flow」と言われるやつ
- この戦略では、主に、5 つのブランチを使用します：
  - main（本番環境用）
  - develop（開発環境用）
  - feature（機能別の作業用）
  - release（リリース準備用）
  - hotfix（緊急バグ修正用）
- これで、ブランチの役割を細かく明確にして、大規模でも安全な開発ができる！
- でも、個人開発や、小規模チームでは、かえって複雑かも、！？
- なので、「GitHub Flow」を採用した
- これは、主に、２つのブランチ（mainと作業用ブランチ）を使用する！
- このシンプルな戦略が、小規模チームには適していると考えました！

って感じの解釈でOKですよ！👍

**Reply by hamanyann** [reply]
ありがとうございます、読みます

**Comment by hamanyann** [comment]
issueテンプレートと、pull requestテンプレートについて理解した。

次、github action ワークフローから。

**Reply by kazzyfrog** [reply]
今回、GitHub Actions を利用するか分からないので、さらっとでいいかもです👍

**Reply by kagomen** [reply]
Note読みましたー！すごい！！！🥳
